src/components/typography/Text.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import styled from '@emotion/styled';
+import { forwardRef, PropsWithChildren } from 'react';
+import { DesignTokensColorsText } from '~types/design-tokens';
+
+export enum TextSize {
+  Small = 'small',
+  Default = 'default',
+  Large = 'large',
+}
+
+interface Props {
+  as?: 'span' | 'p' | 'div';
+  className?: string;
+  colorOverride?: DesignTokensColorsText;
+  isStrong?: boolean;
+  isSubdued?: boolean;
+  size?: TextSize;
+}
+
+export const Text = forwardRef<HTMLSpanElement, PropsWithChildren<Props>>((props, ref) => {
+  const {
+    children,
+    isStrong = false,
+    isSubdued = false,
+    size = TextSize.Default,
+    ...restAndRadixProps
+  } = props;
+
+  return (
+    <StyledText
+      isStrong={isStrong}
+      isSubdued={isSubdued}
+      ref={ref}
+      size={size}
+      {...restAndRadixProps}>
+      {children}
+    </StyledText>
+  );
+});
+Text.displayName = 'Text';
+
+type StyledProps = Required<Pick<Props, 'isSubdued' | 'isStrong' | 'size'>> &
+  Pick<Props, 'colorOverride'>;
+
+const StyledText = styled.span<StyledProps>`
+  max-width: 650px; /* FIXME Use a design token! */
+  font: ${getFontForSizeAndWeight};
+
+  color: ${getColor};
+`;
+
+function getColor({ colorOverride, isSubdued }: Pick<StyledProps, 'colorOverride' | 'isSubdued'>) {
+  if (colorOverride) {
+    return `var(--${colorOverride})`;
+  }
+
+  return isSubdued ? 'var(--echoes-color-text-subdued)' : 'var(--echoes-color-text-default)';
+}
+
+function getFontForSizeAndWeight({ size, isStrong }: Pick<StyledProps, 'size' | 'isStrong'>) {
+  const weight = getFontWeight({ size, isStrong });
+
+  return `var(--echoes-typography-text-${size}-${weight})`;
```

Comment:
I'm not super fan of generating the tokens like that, it makes it hard to find them when there are breaking changes and we need to update them.

Should we just create a map with all the combinations?
Maybe something like this:
```
const TYPOGRAPHY_MAP = {
  [`${TextSize.Small}-false`]: '--echoes-typography-text-small-medium',
  [`${TextSize.Small}-true`]: '--echoes-typography-text-small-semi-bold',
  [`${TextSize.Default}-false`]: '--echoes-typography-text-default-regular',
  [`${TextSize.Default}-true`]: '--echoes-typography-text-default-semi-bold',
  [`${TextSize.Large}-false`]: '--echoes-typography-text-large-regular',
  [`${TextSize.Large}-true`]: '--echoes-typography-text-large-semi-bold',
};

...

font: ${TYPOGRAPHY_MAP[`${size}-${isStrong}`]}
```

Reply:
Yeah, I wasn't sure about this either, let me fix it